core/dive.c

```diff
@@ -1977,20 +2007,20 @@ static char *merge_text(const char *a, const char *b, const char *sep)
 	return res;
 }
 
-#define SORT(a, b, field)         \
-	if (a->field != b->field) \
-		return a->field < b->field ? -1 : 1
+#define SORT(a, b, field, offset)          \
+	if (a->field != b->field + offset) \
+		return a->field < b->field + offset ? -1 : 1
 
-static int sort_event(struct event *a, struct event *b)
+static int sort_event(const struct event *a, const struct event *b, int offset)
 {
-	SORT(a, b, time.seconds);
-	SORT(a, b, type);
-	SORT(a, b, flags);
-	SORT(a, b, value);
+	SORT(a, b, time.seconds, offset);
```

Comment:
This change makes no sense.

Honestly, I see why you do it, but the end result is all kinds of odd. The offset simply doesn't make sense in this context.

Instead of passing in "offset" to sort_event(), just pass in both times. Then this code continues to make sense, and the games with the time will make (more) sense in the context of the caller.

You can see how Dirk is confused about the caller code.